src/lib/alloc.c

```diff
@@ -697,12 +715,65 @@ void rfree(void *ptr)
 	}
 
 	/* free the block */
-	spin_lock_irq(&memmap.lock, flags);
 	free_block(ptr);
-	spin_unlock_irq(&memmap.lock, flags);
 	memmap.heap_trace_updated = 1;
 }
 
+void rfree(void *ptr)
+{
+	uint32_t flags;
+
+	spin_lock_irq(&memmap.lock, flags);
+	_rfree_unlocked(ptr);
+	spin_unlock_irq(&memmap.lock, flags);
+}
+
+void *_realloc(void *ptr, int zone, uint32_t caps, size_t bytes, bool copy)
```

Comment:
Why do we have `copy` parameter? I guess whole point of realloc is to have copy already in it. IMHO We can just assume copy is always true.

Reply:
@jajanusz again asked by @lgirdwood. I'm fine removing it...